### DISCUSSION TITLE
Is it possible to download the vsix file for older versions of jupyter?

### DISCUSSION THREAD

**Discussion by poeschlr** [microsoft/vscode-jupyter#9711] [Questions and Answers] [answered]
I have the issue that my work PC is not connected to the internet and can therefore sadly not directly install the extension from the extension manager.

So i kind of rely on the availability of downloading vsix files. However if i download the file from the market place then it is often not compatible even with the newest release of vscode (like right now one can only download versions 2022.4.xxxxxxxxxxx which do not seem to be compatible with vscode 1.66.2)

**Comment by DonJayamanne** [comment, marked as answer]
If you would like to download a version of the extension that compatible with version 1.66.2, then the easiest is:
* Install vs code 1.66.2
* Download and install Jupyter extension from that version of VS Code
* Keep track of the version of Jupyter installed
* Go to the marketplace and download the VSIX as follows:

Please change the version to match the one you have identified to be the right one.

https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-toolsai/vsextensions/jupyter/2022.4.1001071005/vspackage

**Reply by poeschlr** [reply]
I can see this workaround to be acceptable for outdated versions of VSCode but in my opinion getting the package for the latest version should not require jumping through such hoops.

And so far i really only had an issue with jupyter here. All other extensions either just work if one clicks "download extension". Or they have a separate download page where they handle the usecase for getting the offline installers (like is done for the cpp extension).

**Reply by monafurukawa** [reply]
This method only downloads the alpine-arm64 version, do you know how to download the win32-x64 version?

**Reply by yaogang2060** [reply]
@monafurukawa https://stackoverflow.com/questions/76116113/how-can-i-download-a-vs-code-extension-with-multiple-platforms-via-direct-downlo
